packages/suite/src/components/suite/modals/ReduxModal/UserContextModal/AddAccountModal/AddAccountButton/AddAccountButton.tsx

```diff
@@ -37,66 +36,69 @@ const verifyAvailability = ({
         // discovery failed?
         return 'MODAL_ADD_ACCOUNT_NO_ACCOUNT';
     }
-    if (emptyAccounts.length === 0) {
-        return 'MODAL_ADD_ACCOUNT_NO_EMPTY_ACCOUNT';
-    }
-    if (emptyAccounts.length > 1) {
-        // prev account is empty, do not add another
-        return 'MODAL_ADD_ACCOUNT_PREVIOUS_EMPTY';
-    }
-    if (account.index === 0 && account.empty && account.accountType === 'normal') {
-        // current (first normal) account is empty, do not add another
-        return 'MODAL_ADD_ACCOUNT_PREVIOUS_EMPTY';
-    }
-    if (account.index >= 10) {
-        return 'MODAL_ADD_ACCOUNT_LIMIT_EXCEEDED';
+
+    if (account.networkType !== 'ethereum') {
+        if (emptyAccounts.length === 0) {
+            return 'MODAL_ADD_ACCOUNT_NO_EMPTY_ACCOUNT';
+        }
+        if (emptyAccounts.length > 1) {
+            // prev account is empty, do not add another
+            return 'MODAL_ADD_ACCOUNT_PREVIOUS_EMPTY';
+        }
+        if (account.index === 0 && account.empty && account.accountType === 'normal') {
+            // current (first normal) account is empty, do not add another
+            return 'MODAL_ADD_ACCOUNT_PREVIOUS_EMPTY';
+        }
     }
 };
 
 interface AddAccountButtonProps {
     network: Network;
     selectedAccount?: NetworkAccount;
-    emptyAccounts: Account[];
+    scopedAccounts: Account[];
     onEnableAccount: (account: Account) => void;
+    onAddNewAccount: () => void;
 }
 
 const AddDefaultAccountButton = ({
-    emptyAccounts,
+    scopedAccounts,
     onEnableAccount,
+    onAddNewAccount,
     network,
     selectedAccount,
 }: AddAccountButtonProps) => {
-    const defaultAccount = emptyAccounts[emptyAccounts.length - 1];
+    const defaultAccount = scopedAccounts.at(-1);
     const device = useSelector(selectSelectedDevice);
 
     const { setCoinFilter, setSearchString, coinFilter } = useAccountSearch();
 
     const handleClick = useCallback(() => {
-        const { accountType: type, path, symbol } = defaultAccount;
-        onEnableAccount(defaultAccount);
-        // reset search string in account search box
-        setSearchString(undefined);
-        if (coinFilter && coinFilter !== symbol) {
-            // if coinFilter is active then reset it only if added account doesn't belong to selected/filtered coin
-            setCoinFilter(undefined);
+        if (defaultAccount) {
+            onEnableAccount(defaultAccount);
+            // reset search string in account search box
+            setSearchString(undefined);
+            if (coinFilter && coinFilter !== defaultAccount.symbol) {
+                // if coinFilter is active then reset it only if added account doesn't belong to selected/filtered coin
+                setCoinFilter(undefined);
+            }
+        } else {
+            onAddNewAccount();
```

Comment:
I am thinking that in case you add one account manually, should it one more and make it invisible or not?